src/test/java/io/grpc/annotations/checkers/ExperimentalApiCheckerTest.java

```diff
@@ -0,0 +1,498 @@
+        "  ElementType.PACKAGE,",
+        "  ElementType.TYPE})",
+        "@Documented",
+        "public @interface Internal {}");
+
+    // add the @ExperimentalApi annotation
+    compiler.addSourceLines("io/grpc/ExperimentalApi.java",
+        "package io.grpc;",
+        "",
+        "import java.lang.annotation.Documented;",
+        "import java.lang.annotation.ElementType;",
+        "import java.lang.annotation.Retention;",
+        "import java.lang.annotation.RetentionPolicy;",
+        "import java.lang.annotation.Target;",
+        "",
+        "@Internal",
+        "@Retention(RetentionPolicy.CLASS)",
+        "@Target({",
+        "   ElementType.ANNOTATION_TYPE,",
+        "   ElementType.CONSTRUCTOR,",
+        "   ElementType.FIELD,",
+        "   ElementType.METHOD,",
+        "   ElementType.PACKAGE,",
+        "   ElementType.TYPE})",
+        "@Documented",
+        "public @interface ExperimentalApi {",
+        "  String value() default \"\";",
+        "}");
+
+    // add an annotated class
+    compiler.addSourceLines("io/grpc/AnnotatedClass.java",
+        "package io.grpc;",
+        "",
+        "import io.grpc.ExperimentalApi;",
+        "",
+        "@ExperimentalApi(\"https://example.com/issue\")",
+        "public class AnnotatedClass {",
+        "  public static final int MEMBER = 42;",
+        "  public static int foo() { return 42; }",
+        "}");
+
+    // add a stable api class
+    compiler.addSourceLines("io/grpc/StableApi.java",
+        "package io.grpc;",
+        "",
+        "public class StableApi {",
+        "  public void foo() {}",
+        "}");
+
+    // add a annotated class members;
+    compiler.addSourceLines("io/grpc/AnnotatedMember.java",
+        "package io.grpc;",
+        "",
+        "import io.grpc.ExperimentalApi;",
+        "",
+        "public class AnnotatedMember {",
+        "  @ExperimentalApi",
+        "  public void instanceMethod() {}",
+        "",
+        "  @ExperimentalApi",
+        "  public static int MEMBER = 42;",
+        "",
+        "  @ExperimentalApi",
+        "  public static void staticMethod() {};",
+        "",
+        "",
+        "  @ExperimentalApi",
+        "  public final int member = 42;",
+        "}");
+
+    // add an annotated interface
+    compiler.addSourceLines("io/grpc/IAnnotated.java",
+        "package io.grpc;",
+        "",
+        "import io.grpc.ExperimentalApi;",
+        "",
+        "@ExperimentalApi",
+        "public interface IAnnotated {",
+        "}");
+  }
+
+  @Test
+  public void negative() {
+    compiler
+        .addSourceLines("example/Test.java",
+            "package example;",
+            "",
+            "public class Test {",
+            "  public static void main(String args[]) {",
+            "    System.out.println(args);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  // Test for compilation unit matching
+  @Test
+  public void negativeDefaultPackageCouldEvaluate() {
```

Comment:
I'm not sure what this one is testing - can it be dropped if we are no longer excluding `io.grpc.*` code?

Reply:
Dropped.